model/utils/rng_provider.py

```diff
@@ -5,27 +5,35 @@
 of the simulation changes.
 """
 
+import hashlib
+import logging
 from typing import List, Union
 import numpy as np
-from model.constants import global_rng_entropy
+# from model.constants import global_rng_entropy
 
 class RNGProvider():
     """
     RNGProvider providers randum number generators based
     on a global entropy and contexts
     """
-    entropy: int
+    seed: int
+    monte_carlo_run: int
 
-    def __init__(self, entropy: int):
-        self.entropy = entropy
+    def __init__(self, seed: int, monte_carlo_run: int):
+        self.seed = seed
+        self.monte_carlo_run = monte_carlo_run
 
     def get_rng(self, *context: List[Union[str, int]]) -> np.random.Generator:
-        return np.random.default_rng(self.__seed__(context))
+        return np.random.default_rng(self.__seed__(list(context)))
 
     def __seed__(self, context: List[Union[str, int]]) -> np.random.SeedSequence:
-        return np.random.SeedSequence(
-            self.entropy,
-            spawn_key=map(abs, map(hash, context))
+        seed_sequence = np.random.SeedSequence(
+            self.seed,
+            spawn_key=map(abs, map(__hash__, [self.monte_carlo_run] + context))
```

Comment:
can numbers from the `__hash__` function get negative? Curious how negative values would be encoded / decoded

Reply:
Good catch, that was leftover from the builtin `hash` which needed an abs, removing 👍